assets/js/components/NewChecksResults/ExecutionResults.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import Table from '@components/Table';
+import ReactMarkdown from 'react-markdown';
+import remarkGfm from 'remark-gfm';
+import { ExecutionIcon } from '@components/ExecutionResults/ExecutionIcon';
+import { getCheckDescription } from '@components/ExecutionResults/checksUtils';
+import CheckResultOutline from './CheckResultOutline';
+
+const resultsTableConfig = {
+  usePadding: false,
+  columns: [
+    {
+      title: 'Id',
+      key: 'checkId',
+      render: (checkId) => (
+        <div className="whitespace-nowrap text-jungle-green-500">{checkId}</div>
+      ),
+    },
+    {
+      title: 'Description',
+      key: 'description',
+      render: (description) => (
+        <ReactMarkdown className="markdown" remarkPlugins={[remarkGfm]}>
+          {description}
+        </ReactMarkdown>
+      ),
+    },
+    {
+      title: 'Result',
+      key: 'result',
+      render: (_, { result, executionState }) => (
+        <ExecutionIcon executionState={executionState} health={result} />
+      ),
+    },
+  ],
+  collapsibleDetailRenderer: ({
+    checkId,
+    agentsCheckResults,
+    expectationResults,
+    clusterName,
+  }) => (
+    <CheckResultOutline
+      checkId={checkId}
+      agentsCheckResults={agentsCheckResults}
+      expectationResults={expectationResults}
+      clusterName={clusterName}
+    />
+  ),
+};
+
+const addHostnameToTargets = (targets, hostnames) =>
+  targets?.map((target) => {
+    const { agent_id } = target;
+
+    const { hostname } = hostnames.find(({ id }) => agent_id === id);
+    return {
+      ...target,
+      hostname,
+    };
+  });
+
+function ExecutionResults({
+  clusterName,
+  executionState,
+  checksResults,
+  catalog,
+  hostnames,
+}) {
+  const tableData = checksResults.map(
```

Comment:
Maybe we should have some internal mapper to map the result coming from wanda to our internal object.
Otherwise, it looks like we are pretty much forced to do operations in many places, like adding this hostnames to targets and use variables with `_`.

Reply:
Well, totally fair point. I guess that intervention needs some extra thinking and restructuring of the app.
Food for our frontend guild :smile: